idaes/models_extra/power_generation/costing/power_plant_capcost.py

```diff
@@ -0,0 +1,1995 @@
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+"""
+Power Plant costing library
+This method leverages NETL costing capabilities. Two main methods have been
+developed to calculate the capital cost of power generation plants:
+
+1. Fossil fueled power plants (from SCPC to IGCC) (get_PP_costing)
+2. Supercritical CO2 power cycles (direct and indirect) (get_sCO2_unit_cost)
+
+Other methods:
+
+    - get_fixed_OM_costs() to cost fixed O&M costs
+    - get_variable_OM_costs() to cost variable O&M costs
+    - get_ASU_cost() to cost air separation units
+    - costing_initialization() to initialize costing blocks
+    - display_total_plant_costs() to display total plant cost (TPC)
+    - display_bare_erected_costs() to display BEC costs
+    - get_total_TPC() to display the total TPC of the entire flowsheet
+    - display_flowsheet_cost() to display flowsheet cost
+    - check_sCO2_costing_bounds() to display a warning if costing model have
+      been used outside the range that where designed for
+"""
+__author__ = "Costing Team (A. Noring, B. Paul, D. Caballero, and M. Zamarripa)"
+__version__ = "1.0.0"
+
+from pandas import DataFrame
+from sys import stdout
+import textwrap
+
+from pyomo.environ import (
+    Param,
+    Var,
+    Block,
+    Constraint,
+    Expression,
+    value,
+    Expr_if,
+    units as pyunits,
+)
+from pyomo.core.base.expression import ScalarExpression
+from pyomo.core.base.units_container import InconsistentUnitsError
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+import idaes.core.util.scaling as iscale
+from idaes.core import register_idaes_currency_units
+from idaes.models_extra.power_generation.costing.costing_dictionaries import (
+    BB_costing_params,
+    sCO2_costing_params,
+)
+
+from idaes.core import (
+    FlowsheetCostingBlockData,
+    register_idaes_currency_units,
+)
+
+from idaes.core.util.tables import stream_table_dataframe_to_string
+
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+# Register standard currency units
+register_idaes_currency_units()
+
+# -----------------------------------------------------------------------------
+# Power Plant Costing Library
+# -----------------------------------------------------------------------------
+
+import pyomo.environ as pyo
+from idaes.core import declare_process_block_class
+
+
+@declare_process_block_class("QGESSCosting")
+class QGESSCostingData(FlowsheetCostingBlockData):
+    # Register currency and conversion rates based on CE Index
+    register_idaes_currency_units()
```

Comment:
If `register_idaes_currency_units()` is already called when this module is loaded is there any reason to call it here?

Reply:
`register_idaes_currency_units` should not be called at the module level.